group-coordinator/src/main/java/org/apache/kafka/coordinator/group/consumer/ConsumerGroup.java

```diff
@@ -423,6 +456,47 @@ public Map<String, TopicMetadata> computeSubscriptionMetadata(
         return Collections.unmodifiableMap(newSubscriptionMetadata);
     }
 
+    /**
+     * Updates the next metadata refresh time.
+     *
+     * @param nextTimeMs The next time in milliseconds.
+     * @param groupEpoch The associated group epoch.
+     */
+    public void setNextMetadataRefreshTime(
+        long nextTimeMs,
+        int groupEpoch
+    ) {
+        this.nextMetadataRefreshTime = new TimeAndEpoch(nextTimeMs, groupEpoch);
+    }
+
+    /**
+     * Resets the next metadata refresh.
+     */
+    public void resetNextMetadataRefreshTime() {
+        this.nextMetadataRefreshTime = TimeAndEpoch.EMPTY;
+    }
+
+    /**
+     * Checks if a metadata refresh is required. A refresh is required in two cases:
+     * 1) The next update time is smaller or equals to the current time;
+     * 2) The group epoch associated with the next update time is smaller than
```

Comment:
This is also the case when we reset the refreshMetadata time right?

Reply:
Right. In this case, the time is set to zero so it is always smaller than the current time.

Reply:
shouldn't it be "associated with the next update time is larger than"?

the "current group epoch" is `groupEpoch()` right